pkg/s3select/input.go

```diff
@@ -75,6 +80,143 @@ type Input struct {
 	stats           *statInfo
 }
 
+// JSONInput represents a record producing input from a  formatted file or pipe.
+type JSONInput struct {
+	options         *JSONOptions
+	reader          *jsoniter.Decoder
+	firstRow        []string
+	header          []string
+	minOutputLength int
+	stats           *statInfo
+}
+
+// ParseSelectTokens tokenizes the select query into required Columns, Alias, limit value
+// where clause, aggregate functions, myFunctions, error.
+type ParseSelectTokens struct {
+	reqCols          []string
+	alias            string
+	myLimit          int64
+	whereClause      interface{}
+	aggFunctionNames []string
+	myFuncs          *SelectFuncs
+	myErr            error
+}
+
+// SelectQuery Interface has methods implemented by csv and JSON
+type SelectQuery interface {
+	Header() []string
+	hasHeader() bool
+	getHeader() []string
+	getOutPutFieldDelimitter() string
+	getBytesProcessed(individualRecord map[string]interface{})
+	Read() map[string]interface{}
+	getExpression() string
+	bytesReturned(int64)
+	createStatXML() (string, error)
+	createProgressXML() (string, error)
+	evaluateFuncErr(myVal *sqlparser.FuncExpr) error
+	colNameErrs(columnNames []string) error
+	getProgressFlag() bool
+}
+
+func (reader *Input) getProgressFlag() bool {
+	return reader.options.Progress
+}
+func (reader *JSONInput) getProgressFlag() bool {
+	return reader.options.Progress
+}
+
+func (reader *Input) getBytesProcessed(individualRecord map[string]interface{}) {
+	// Do it for now but we have to take the value not the keys for CSV
+	out, err := json.Marshal(individualRecord)
+	if err != nil {
+		fmt.Println("error:", err)
```

Comment:
please remove all `println` s

Reply:
Also, it will be cleaner to read if you can separate out the implementations of both CSV and JSON and not interleave them.

Reply:
```println``` removed . For Json and CSV file , we can have it as different files .Will do it in another PR